dashboard/test/models/contact_rollups_raw_test.rb

```diff
@@ -0,0 +1,22 @@
+require 'test_helper'
+
+class ContactRollupsRawTest < ActiveSupport::TestCase
+  test 'extract_email_preferences adds all email preferences' do
+    email_preference = create :email_preference
+    ContactRollupsRaw.extract_email_preferences
+
+    expected_data = {opt_in: email_preference.opt_in}
+    result = ContactRollupsRaw.find_by(
+      email: email_preference.email,
+      sources: "dashboard.#{email_preference.class.table_name}"
+    )
+
+    assert_equal expected_data, result.data.symbolize_keys
+  end
+
+  test 'bunch of email prefences works' do
+    3.times {|i| create :email_preference, email: "contact_#{i}@rollups.com"}
+    ContactRollupsRaw.extract_email_preferences
+    assert 3, ContactRollupsRaw.count
```

Comment:
You also want to do `ContactRollupsRaw.delete_all` to assure the table is empty before inserting to it.

Reply:
Isn't that an assumption of our test framework (cleanup/teardown of database before/after every test)?

Reply:
Good question. I don't know what is our current test cleaning strategy and how it behaves in local dev machine, in Drone and in test machine. Would be very helpful for me and the team if you could do some research and share what you find.